tools/fifo.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 Codethink Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fifo.h"
+
+uint16_t fifo_advance(uint16_t pos)
+{
+	pos++;
+	return (pos >= FIFO_MAX) ? 0 : pos;
+}
+
+uint16_t fifo_seek(int16_t pos, int16_t steps)
```

Comment:
Maybe bump these to 32-bit because `pos` may by up to `UINT16_MAX`, and steps may be up to +/- `UINT16_MAX`.

```suggestion
uint16_t fifo_seek(int32_t pos, int32_t steps)
```